### PR TITLE
[TACHYON-304] Add worker WebUI

### PR DESCRIPTION
**Pull request by carsonwang**
We proposed to add webUI for the workers. It displays the detailed hierarchy storage details, including all storage dirs. This would be helpful for admins to check the worker's configuration and debug issues. Please see the attached screenshot.
A follow-up PR will link the master WebUI to the workers' WebUI.
workerwebui

### CODE REVIEW

File: core/src/main/java/tachyon/web/UIWebServer.java [hunk 1]
```diff
@@ -83,30 +83,55 @@ public UIWebServer(String serverName, InetSocketAddress address, MasterInfo mast
...
+  /**
+   * Setup the servlets for the master web server.
+   *
+   * @param masterInfo The MasterInfo instance in the master
+   */
+  public void setupMasterWebServer(MasterInfo masterInfo) {
+    Preconditions.checkNotNull(masterInfo, "Master information cannot be null");
+
+    mWebAppContext.addServlet(new ServletHolder(new WebInterfaceGeneralServlet(masterInfo)),
+            "/home");
```
Comment (calvinjia) [inline core/src/main/java/tachyon/web/UIWebServer.java]: Too much indentation

**Comment by calvinjia** [comment]
Do you think it would make more sense to have a base class UIWebServer and have Master/Worker servers inherit from it?

**Comment by carsonwang** [comment]
Hi @calvinjia , thanks for the suggestion. I just pushed an update that make UIWebServer a base class and have Master/Worker servers inherit from it. The code style is fixed as well.

File: core/src/main/java/tachyon/worker/WorkerStorage.java [hunk 2]
```diff
@@ -589,6 +592,53 @@ public void freeBlocks(List<Long> blockIds) {
...
+   * Get the capacity of the worker.
+   *
+   * @return the worker's capacity in bytes.
+   */
+  public long getCapacityBytes() {
+    return mCapacityBytes;
+  }
+
+  /**
+   * Get the capacity of the MEM/SSH/HDD tier.
```
Comment (calvinjia) [inline core/src/main/java/tachyon/worker/WorkerStorage.java]: Should this be MEM/SSD/HDD?

File: core/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        CommonUtils.getSizeFromBytes(mWorkerStorage.getCapacityBytes(StorageLevelAlias.SSD)));
+
+    request.setAttribute("ssdUsedBytes",
+        CommonUtils.getSizeFromBytes(mWorkerStorage.getUsedBytes(StorageLevelAlias.SSD)));
+
+    request.setAttribute("hddCapacityBytes",
+        CommonUtils.getSizeFromBytes(mWorkerStorage.getCapacityBytes(StorageLevelAlias.HDD)));
+
+    request.setAttribute("hddUsedBytes",
+        CommonUtils.getSizeFromBytes(mWorkerStorage.getUsedBytes(StorageLevelAlias.HDD)));
```
Comment (calvinjia) [inline core/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java]: Could the storage alias info be derived from the storageDirs by iterating all values of the enum? This way will not work for custom storage aliases right?

**Comment by swatDong** [comment]
In UIWebServer.java, it still use `Constants.MASTER_WEB_THREAD_COUNT
` as the default thread number. Should Master and Worker use the different ones?

File: core/src/main/java/tachyon/worker/WorkerStorage.java (hunk 2 above)
Comment (carsonwang) [inline core/src/main/java/tachyon/worker/WorkerStorage.java]: Good catch. I'll fix it. Thanks.

File: core/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java (hunk 3 above)
Comment (carsonwang) [inline core/src/main/java/tachyon/web/WebInterfaceWorkerGeneralServlet.java]: Ok, I am going to reuse the getUsedBytesOnTiers() in WorkerStorage. So we don't need hard code the mem/ssd/hdd here.

**Comment by carsonwang** [comment]
@swatDong , good catch! Thank. I suppose the master and the worker can share the same web thread count setting. I will rename this to WEB_THREAD_COUNT.

**Comment by carsonwang** [comment]
I just updated and rebased the code to address the comments above. Thanks!

**Comment by calvinjia** [comment]
+1 LGTM

**Comment by haoyuan** [comment]
Thanks!

**Pull request merged**